<?xml version="1.0" encoding="UTF-8"?>
<testng-results retried="9" ignored="0" total="18" passed="6" failed="3" skipped="0">
  <reporter-output>
  </reporter-output>
  <suite started-at="2024-12-28T11:47:28 IST" name="Web Suite" finished-at="2024-12-28T11:50:16 IST" duration-ms="168578">
    <groups>
    </groups>
    <test started-at="2024-12-28T11:47:28 IST" name="Web Tests" finished-at="2024-12-28T11:50:16 IST" duration-ms="168578">
      <class name="sanity.ShopDemo">
        <test-method is-config="true" signature="startSession(java.lang.String)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:47:28 IST" name="startSession" finished-at="2024-12-28T11:47:54 IST" duration-ms="26441" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[web]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- startSession -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:47:54 IST" name="beforeMethod" finished-at="2024-12-28T11:47:55 IST" duration-ms="620" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test01_verifySearchForSpecificProduct(java.lang.String,java.lang.String)]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method signature="test01_verifySearchForSpecificProduct(java.lang.String,java.lang.String)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:47:56 IST" name="test01_verifySearchForSpecificProduct" description="Test01 - Verify search for a specific product" data-provider="specificProductData" finished-at="2024-12-28T11:48:00 IST" duration-ms="4635" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[Anchor Bracelet]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[Anchor Bracelet]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test01_verifySearchForSpecificProduct -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:01 IST" name="afterMethod" finished-at="2024-12-28T11:48:01 IST" duration-ms="377" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:01 IST" name="beforeMethod" finished-at="2024-12-28T11:48:01 IST" duration-ms="24" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test01_verifySearchForSpecificProduct(java.lang.String,java.lang.String)]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method signature="test01_verifySearchForSpecificProduct(java.lang.String,java.lang.String)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:01 IST" name="test01_verifySearchForSpecificProduct" description="Test01 - Verify search for a specific product" data-provider="specificProductData" finished-at="2024-12-28T11:48:04 IST" duration-ms="2726" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[ATID Yellow Shoes]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[ATID Yellow Shoes]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test01_verifySearchForSpecificProduct -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:04 IST" name="afterMethod" finished-at="2024-12-28T11:48:04 IST" duration-ms="642" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:05 IST" name="beforeMethod" finished-at="2024-12-28T11:48:05 IST" duration-ms="33" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test02_SearchForKeywordAndVerifyProductsCount(java.lang.String,java.lang.String)]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method signature="test02_SearchForKeywordAndVerifyProductsCount(java.lang.String,java.lang.String)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:05 IST" name="test02_SearchForKeywordAndVerifyProductsCount" description="Test02 - Verify search for a keyword and validate the number of products in the results" data-provider="searchKeywordData" finished-at="2024-12-28T11:48:07 IST" duration-ms="2213" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[shoes]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[5]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test02_SearchForKeywordAndVerifyProductsCount -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:07 IST" name="afterMethod" finished-at="2024-12-28T11:48:08 IST" duration-ms="375" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:08 IST" name="beforeMethod" finished-at="2024-12-28T11:48:08 IST" duration-ms="23" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test02_SearchForKeywordAndVerifyProductsCount(java.lang.String,java.lang.String)]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method signature="test02_SearchForKeywordAndVerifyProductsCount(java.lang.String,java.lang.String)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:08 IST" name="test02_SearchForKeywordAndVerifyProductsCount" description="Test02 - Verify search for a keyword and validate the number of products in the results" data-provider="searchKeywordData" finished-at="2024-12-28T11:48:11 IST" duration-ms="2706" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[jeans]]>
              </value>
            </param>
            <param index="1">
              <value>
                <![CDATA[7]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test02_SearchForKeywordAndVerifyProductsCount -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:11 IST" name="afterMethod" finished-at="2024-12-28T11:48:11 IST" duration-ms="482" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:11 IST" name="beforeMethod" finished-at="2024-12-28T11:48:11 IST" duration-ms="24" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test03_verifyAboutAtidText()]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method signature="test03_verifyAboutAtidText()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:11 IST" name="test03_verifyAboutAtidText" description="Test03 - Verify About Atid Text" finished-at="2024-12-28T11:48:13 IST" duration-ms="1452" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test03_verifyAboutAtidText -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:13 IST" name="afterMethod" finished-at="2024-12-28T11:48:13 IST" duration-ms="503" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:13 IST" name="beforeMethod" finished-at="2024-12-28T11:48:13 IST" duration-ms="22" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test04_verifyProductImage()]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method retried="true" signature="test04_verifyProductImage()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:13 IST" name="test04_verifyProductImage" description="Test04 - Verify product Image" finished-at="2024-12-28T11:48:20 IST" duration-ms="6431" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Error Comparing Image File: FindFailed: C:\\Automation\\Projects\\Selenium4Java\\Images\\anchorBraceletImage.png: (156x156) in R[0,0 1280x1024]@S(0)
  Line 2287, in file Region.java
]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Error Comparing Image File: FindFailed: C:\\Automation\\Projects\\Selenium4Java\\Images\\anchorBraceletImage.png: (156x156) in R[0,0 1280x1024]@S(0)
  Line 2287, in file Region.java

at org.testng.Assert.fail(Assert.java:99)
at extensions.Verifications.visualElement(Verifications.java:185)
at sanity.ShopDemo.test04_verifyProductImage(ShopDemo.java:50)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test04_verifyProductImage -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:20 IST" name="afterMethod" finished-at="2024-12-28T11:48:20 IST" duration-ms="608" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:20 IST" name="beforeMethod" finished-at="2024-12-28T11:48:21 IST" duration-ms="143" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test04_verifyProductImage()]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method retried="true" signature="test04_verifyProductImage()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:21 IST" name="test04_verifyProductImage" description="Test04 - Verify product Image" finished-at="2024-12-28T11:48:23 IST" duration-ms="2839" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Error Comparing Image File: FindFailed: C:\\Automation\\Projects\\Selenium4Java\\Images\\anchorBraceletImage.png: (156x156) in R[0,0 1280x1024]@S(0)
  Line 2287, in file Region.java
]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Error Comparing Image File: FindFailed: C:\\Automation\\Projects\\Selenium4Java\\Images\\anchorBraceletImage.png: (156x156) in R[0,0 1280x1024]@S(0)
  Line 2287, in file Region.java

at org.testng.Assert.fail(Assert.java:99)
at extensions.Verifications.visualElement(Verifications.java:185)
at sanity.ShopDemo.test04_verifyProductImage(ShopDemo.java:50)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.retryFailed(TestInvoker.java:213)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:58)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test04_verifyProductImage -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:23 IST" name="afterMethod" finished-at="2024-12-28T11:48:24 IST" duration-ms="736" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:24 IST" name="beforeMethod" finished-at="2024-12-28T11:48:24 IST" duration-ms="149" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test04_verifyProductImage()]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method retried="true" signature="test04_verifyProductImage()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:24 IST" name="test04_verifyProductImage" description="Test04 - Verify product Image" finished-at="2024-12-28T11:48:27 IST" duration-ms="2906" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Error Comparing Image File: FindFailed: C:\\Automation\\Projects\\Selenium4Java\\Images\\anchorBraceletImage.png: (156x156) in R[0,0 1280x1024]@S(0)
  Line 2287, in file Region.java
]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Error Comparing Image File: FindFailed: C:\\Automation\\Projects\\Selenium4Java\\Images\\anchorBraceletImage.png: (156x156) in R[0,0 1280x1024]@S(0)
  Line 2287, in file Region.java

at org.testng.Assert.fail(Assert.java:99)
at extensions.Verifications.visualElement(Verifications.java:185)
at sanity.ShopDemo.test04_verifyProductImage(ShopDemo.java:50)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.retryFailed(TestInvoker.java:213)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:58)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test04_verifyProductImage -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:28 IST" name="afterMethod" finished-at="2024-12-28T11:48:29 IST" duration-ms="881" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:30 IST" name="beforeMethod" finished-at="2024-12-28T11:48:31 IST" duration-ms="348" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test04_verifyProductImage()]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method signature="test04_verifyProductImage()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:31 IST" name="test04_verifyProductImage" description="Test04 - Verify product Image" finished-at="2024-12-28T11:48:34 IST" duration-ms="3806" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Error Comparing Image File: FindFailed: C:\\Automation\\Projects\\Selenium4Java\\Images\\anchorBraceletImage.png: (156x156) in R[0,0 1280x1024]@S(0)
  Line 2287, in file Region.java
]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Error Comparing Image File: FindFailed: C:\\Automation\\Projects\\Selenium4Java\\Images\\anchorBraceletImage.png: (156x156) in R[0,0 1280x1024]@S(0)
  Line 2287, in file Region.java

at org.testng.Assert.fail(Assert.java:99)
at extensions.Verifications.visualElement(Verifications.java:185)
at sanity.ShopDemo.test04_verifyProductImage(ShopDemo.java:50)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.retryFailed(TestInvoker.java:213)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:58)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test04_verifyProductImage -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:36 IST" name="afterMethod" finished-at="2024-12-28T11:48:37 IST" duration-ms="795" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:37 IST" name="beforeMethod" finished-at="2024-12-28T11:48:37 IST" duration-ms="367" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test05_verifyHighestProductPrice()]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method retried="true" signature="test05_verifyHighestProductPrice()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:37 IST" name="test05_verifyHighestProductPrice" description="Test05 - Verify Highest Product Price" finished-at="2024-12-28T11:48:41 IST" duration-ms="3727" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertTrue(Assert.java:45)
at org.testng.Assert.assertTrue(Assert.java:55)
at extensions.Verifications.verifyTextContainedInElement(Verifications.java:110)
at sanity.ShopDemo.test05_verifyHighestProductPrice(ShopDemo.java:59)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test05_verifyHighestProductPrice -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:41 IST" name="afterMethod" finished-at="2024-12-28T11:48:42 IST" duration-ms="1501" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:43 IST" name="beforeMethod" finished-at="2024-12-28T11:48:43 IST" duration-ms="593" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test05_verifyHighestProductPrice()]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method retried="true" signature="test05_verifyHighestProductPrice()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:43 IST" name="test05_verifyHighestProductPrice" description="Test05 - Verify Highest Product Price" finished-at="2024-12-28T11:48:47 IST" duration-ms="4320" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertTrue(Assert.java:45)
at org.testng.Assert.assertTrue(Assert.java:55)
at extensions.Verifications.verifyTextContainedInElement(Verifications.java:110)
at sanity.ShopDemo.test05_verifyHighestProductPrice(ShopDemo.java:59)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.retryFailed(TestInvoker.java:213)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:58)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test05_verifyHighestProductPrice -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:47 IST" name="afterMethod" finished-at="2024-12-28T11:48:49 IST" duration-ms="1383" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:49 IST" name="beforeMethod" finished-at="2024-12-28T11:48:50 IST" duration-ms="717" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test05_verifyHighestProductPrice()]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method retried="true" signature="test05_verifyHighestProductPrice()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:50 IST" name="test05_verifyHighestProductPrice" description="Test05 - Verify Highest Product Price" finished-at="2024-12-28T11:48:55 IST" duration-ms="5363" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertTrue(Assert.java:45)
at org.testng.Assert.assertTrue(Assert.java:55)
at extensions.Verifications.verifyTextContainedInElement(Verifications.java:110)
at sanity.ShopDemo.test05_verifyHighestProductPrice(ShopDemo.java:59)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.retryFailed(TestInvoker.java:213)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:58)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test05_verifyHighestProductPrice -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:55 IST" name="afterMethod" finished-at="2024-12-28T11:48:57 IST" duration-ms="1608" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:57 IST" name="beforeMethod" finished-at="2024-12-28T11:48:58 IST" duration-ms="1071" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test05_verifyHighestProductPrice()]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method signature="test05_verifyHighestProductPrice()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:48:58 IST" name="test05_verifyHighestProductPrice" description="Test05 - Verify Highest Product Price" finished-at="2024-12-28T11:49:05 IST" duration-ms="6333" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:99)
at org.testng.Assert.failNotEquals(Assert.java:1037)
at org.testng.Assert.assertTrue(Assert.java:45)
at org.testng.Assert.assertTrue(Assert.java:55)
at extensions.Verifications.verifyTextContainedInElement(Verifications.java:110)
at sanity.ShopDemo.test05_verifyHighestProductPrice(ShopDemo.java:59)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.retryFailed(TestInvoker.java:213)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:58)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test05_verifyHighestProductPrice -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:05 IST" name="afterMethod" finished-at="2024-12-28T11:49:07 IST" duration-ms="1687" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:07 IST" name="beforeMethod" finished-at="2024-12-28T11:49:08 IST" duration-ms="642" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test06_verifyDetailsOfProductAddedToCart() throws java.lang.Exception]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method signature="test06_verifyDetailsOfProductAddedToCart()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:08 IST" name="test06_verifyDetailsOfProductAddedToCart" description="Test06 - Verify Details Of Product Added To Cart" finished-at="2024-12-28T11:49:27 IST" duration-ms="19222" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test06_verifyDetailsOfProductAddedToCart -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:27 IST" name="afterMethod" finished-at="2024-12-28T11:49:30 IST" duration-ms="2829" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:30 IST" name="beforeMethod" finished-at="2024-12-28T11:49:31 IST" duration-ms="56" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test07_verifySearchProductsElementLocation()]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method retried="true" signature="test07_verifySearchProductsElementLocation()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:31 IST" name="test07_verifySearchProductsElementLocation" description="Test07 - Verify Search Products Element Location" finished-at="2024-12-28T11:49:34 IST" duration-ms="3670" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[The following asserts failed:
expected [198] but found [217],
expected [43] but found [42]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: The following asserts failed:
expected [198] but found [217],
expected [43] but found [42]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:47)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:31)
at extensions.Verifications.verifyElementOrientaion(Verifications.java:269)
at sanity.ShopDemo.test07_verifySearchProductsElementLocation(ShopDemo.java:116)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test07_verifySearchProductsElementLocation -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:34 IST" name="afterMethod" finished-at="2024-12-28T11:49:37 IST" duration-ms="2526" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:37 IST" name="beforeMethod" finished-at="2024-12-28T11:49:37 IST" duration-ms="274" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test07_verifySearchProductsElementLocation()]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method retried="true" signature="test07_verifySearchProductsElementLocation()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:37 IST" name="test07_verifySearchProductsElementLocation" description="Test07 - Verify Search Products Element Location" finished-at="2024-12-28T11:49:41 IST" duration-ms="3206" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[The following asserts failed:
expected [198] but found [217],
expected [43] but found [42],
expected [198] but found [217],
expected [43] but found [42]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: The following asserts failed:
expected [198] but found [217],
expected [43] but found [42],
expected [198] but found [217],
expected [43] but found [42]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:47)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:31)
at extensions.Verifications.verifyElementOrientaion(Verifications.java:269)
at sanity.ShopDemo.test07_verifySearchProductsElementLocation(ShopDemo.java:116)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.retryFailed(TestInvoker.java:213)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:58)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test07_verifySearchProductsElementLocation -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:42 IST" name="afterMethod" finished-at="2024-12-28T11:49:44 IST" duration-ms="2081" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:44 IST" name="beforeMethod" finished-at="2024-12-28T11:49:46 IST" duration-ms="1551" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test07_verifySearchProductsElementLocation()]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method retried="true" signature="test07_verifySearchProductsElementLocation()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:46 IST" name="test07_verifySearchProductsElementLocation" description="Test07 - Verify Search Products Element Location" finished-at="2024-12-28T11:49:50 IST" duration-ms="4340" status="SKIP">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[The following asserts failed:
expected [198] but found [217],
expected [43] but found [42],
expected [198] but found [217],
expected [43] but found [42],
expected [198] but found [217],
expected [43] but found [42]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: The following asserts failed:
expected [198] but found [217],
expected [43] but found [42],
expected [198] but found [217],
expected [43] but found [42],
expected [198] but found [217],
expected [43] but found [42]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:47)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:31)
at extensions.Verifications.verifyElementOrientaion(Verifications.java:269)
at sanity.ShopDemo.test07_verifySearchProductsElementLocation(ShopDemo.java:116)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.retryFailed(TestInvoker.java:213)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:58)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test07_verifySearchProductsElementLocation -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:50 IST" name="afterMethod" finished-at="2024-12-28T11:49:53 IST" duration-ms="2855" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="beforeMethod(java.lang.reflect.Method)[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:53 IST" name="beforeMethod" finished-at="2024-12-28T11:49:55 IST" duration-ms="1285" status="PASS">
          <params>
            <param index="0">
              <value>
                <![CDATA[public void sanity.ShopDemo.test07_verifySearchProductsElementLocation()]]>
              </value>
            </param>
          </params>
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeMethod -->
        <test-method signature="test07_verifySearchProductsElementLocation()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:49:55 IST" name="test07_verifySearchProductsElementLocation" description="Test07 - Verify Search Products Element Location" finished-at="2024-12-28T11:50:00 IST" duration-ms="5528" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[The following asserts failed:
expected [198] but found [217],
expected [43] but found [42],
expected [198] but found [217],
expected [43] but found [42],
expected [198] but found [217],
expected [43] but found [42],
expected [198] but found [217],
expected [43] but found [42]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: The following asserts failed:
expected [198] but found [217],
expected [43] but found [42],
expected [198] but found [217],
expected [43] but found [42],
expected [198] but found [217],
expected [43] but found [42],
expected [198] but found [217],
expected [43] but found [42]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:47)
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:31)
at extensions.Verifications.verifyElementOrientaion(Verifications.java:269)
at sanity.ShopDemo.test07_verifySearchProductsElementLocation(ShopDemo.java:116)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:133)
at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:598)
at org.testng.internal.TestInvoker.retryFailed(TestInvoker.java:213)
at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:58)
at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
at org.testng.TestRunner.privateRun(TestRunner.java:794)
at org.testng.TestRunner.run(TestRunner.java:596)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
at org.testng.SuiteRunner.run(SuiteRunner.java:276)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
at org.testng.TestNG.runSuites(TestNG.java:1063)
at org.testng.TestNG.run(TestNG.java:1031)
at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test07_verifySearchProductsElementLocation -->
        <test-method is-config="true" signature="afterMethod()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:50:02 IST" name="afterMethod" finished-at="2024-12-28T11:50:04 IST" duration-ms="2363" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterMethod -->
        <test-method is-config="true" signature="closeSession()[pri:0, instance:sanity.ShopDemo@c6e0f32]" started-at="2024-12-28T11:50:04 IST" name="closeSession" finished-at="2024-12-28T11:50:16 IST" duration-ms="12143" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeSession -->
      </class> <!-- sanity.ShopDemo -->
    </test> <!-- Web Tests -->
  </suite> <!-- Web Suite -->
</testng-results>
